esx_service/vmdk_ops.py

```diff
@@ -640,6 +643,21 @@ def dev_info(unit_number, pci_slot_number):
     return {'Unit': str(unit_number),
             'ControllerPciSlotNumber': pci_slot_number}
 
+def reset_vol_meta(vmdk_path):
+    '''Clears metadata for vmdk_path'''
+    vol_meta = kv.getAll(vmdk_path)
+    if not vol_meta:
+        vol_meta = {}
+    logging.debug("Reseting meta-data for disk=%s", vmdk_path)
+    if vol_meta.keys() & {kv.STATUS, kv.ATTACHED_VM_UUID, kv.ATTACHED_VM_NAME}:
```

Comment:
Seems to be a 2.7 vs.  3.5 Python issue, I checked on both and am able to simulate on the 2.7 version:

$ python
Python 2.7.6 (default, Mar 22 2014, 22:59:56)
>>> if x.keys() & {'status', 'attached_vm'}:
...      print(x)
...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for &: 'list' and 'set'
>>>

On 3.5
 python
Python 3.5.1+ (default, Apr  4 2016, 12:10:58)
>>> if x.keys() & {'status', 'attached_vm'}:
...     print(x)
... ^D
>>>